src/JohnDeere/Server.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace SocialiteProviders\JohnDeere;
+
+use League\OAuth1\Client\Credentials\TemporaryCredentials;
+use League\OAuth1\Client\Credentials\TokenCredentials;
+use SocialiteProviders\Manager\OAuth1\Server as BaseServer;
+use SocialiteProviders\Manager\OAuth1\User;
+
+class Server extends BaseServer
+{
+    const SANDBOX_URL = 'https://sandboxapi.deere.com';
+
+    /**
+     * oauth_verifier stored for use with.
+     *
+     * @var string
+     */
+    protected $verifier;
+
+    /**
+     * Get the base URL from services config file
+     * 
+     * @return string
+     */
+    protected function baseUrl()
+    {
+        return config('services.john-deere.url', self::SANDBOX_URL);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function urlTemporaryCredentials()
+    {
+        return $this->baseUrl().'platform/oauth/request_token';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function urlAuthorization()
+    {
+        return 'https://my.deere.com/consentToUseOfData';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function urlTokenCredentials()
+    {
+        return $this->baseUrl().'platform/oauth/access_token';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function urlUserDetails()
+    {
+        return $this->baseUrl().'platform/users/@currentUser';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTokenCredentials(TemporaryCredentials $temporaryCredentials, $temporaryIdentifier, $verifier)
+    {
+        $this->verifier = $verifier;
+
+        return parent::getTokenCredentials($temporaryCredentials, $temporaryIdentifier, $verifier);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function userDetails($data, TokenCredentials $tokenCredentials)
+    {
+        $user = new User();
+        $user->id = $data['accountName'];
+        $user->name = "{$data['givenName']} {$data['familyName']}";
+
+        $used = ['accountName', 'givenName', 'familyName'];
```

Comment:
Maybe there's a better name for this variable to make it more clear what you're doing here